pkg/vtpm/Dockerfile

```diff
@@ -70,6 +70,17 @@ FROM scratch
 
 COPY --from=build /out/ /
 COPY init.sh /usr/bin/
+
+# HACK BEGIN
```

Comment:
@deitch do you have an example of Linuxkit containers running with a particular user and group id?

Reply:
ping @deitch

Reply:
Can't you configure it in the json file? - https://github.com/linuxkit/linuxkit/blob/master/pkg/init/cmd/service/prepare.go#L35

Reply:
@christoph-zededa looks like that is to use/create a user namespace, and not a user. Let me ping @deitch offline.

Reply:
I will take a look. Sorry for not responding earlier; GitHub alert overload makes me miss a few things.

Reply:
I did set the uid/gid for vtpm in `rootfs.yaml`, but for some reason containerd failed to even start the container after this change! didn't dig into it to see why.

Reply:
I was able to replicate it, albeit I am not sure that this is the issue. I took the [minimal example](https://github.com/linuxkit/linuxkit/blob/master/examples/minimal.yml) add added `uid: 100`, so it is:

```json
services:
  - name: getty
    image: linuxkit/getty:06f34bce0facea79161566d67345c3ea49965437
    uid: 100
    env:
     - INSECURE=true
```

I then built it. I also did the same without the change.

The built images are identical, except for `containers/services/getty/config.json`, which has the diff on the very first lines:

```diff
5c5
<             "uid": 0,
---
>             "uid": 100,
```

So:

```json
    "process": {
        "user": {
            "uid": 0,
            "gid": 0
        },
```

That just controls how the process is given a UID. So the build process definitely is doing the right thing. Yet I do not get a getty when I start it.

Could it be that vtpm just needs permissions to be non-root? Or has something that checks if it starts as non-root?

I am going to run a similar example. but with nginx, so I can see what is running.

Reply:
@deitch 

> Could it be that vtpm just needs permissions to be non-root?

No this is not the case, changes in the current PR makes vtpm setuid non-root  and succeeds to run it.

> Or has something that checks if it starts as non-root?

same as above.

Reply:
> ctr: cannot exec in a stopped state: unknown

Yeah, that is it. It means that the process started and stopped. Can you see what the logs are? Should be something like `/var/log/vtpm.*` (one file for stdout, one for stderr). Let's see what it is complaining about.

Reply:
Both `/var/log/` and `/containers/services/vtpm/rootfs/var/log/` are empty, and no trace of vtpm in `/persist/newlog/collect/*` either.

Reply:
There was some logic we added a while back to collect containerd logs, in case it is containerd itself that is having an issue. That often scrolls by on console without a chance to see it. I will see if I can find it.

Reply:
Ah, yes, here it is. `vtpm` is under `services:` and not `onboot:`, right? So containerd is running it. 

See if this helps, build an image with modified containerd logging and see if that gives anything.

Reply:
I moved it up to the `onboot`, now eve just freeze trying to execute it, I'm not acquainted with linuxkit, but tried to export it and load in docker and this happens:
```
make IMAGE=lfedge/eve-vtpm:8979f64dd03931c6a6453b7b30ba61e2ad69ef28-dirty-95d5ec3 cache-export-docker-load
docker run -it lfedge/eve-vtpm:8979f64dd03931c6a6453b7b30ba61e2ad69ef28-dirty-95d5ec3
```
and : 
```
docker ps
CONTAINER ID   IMAGE                                                                    COMMAND                  CREATED              STATUS              PORTS     NAMES
37af2d2fb908   lfedge/eve-vtpm:8979f64dd03931c6a6453b7b30ba61e2ad69ef28-dirty-95d5ec3   "/usr/bin/init.sh"       About a minute ago   Up About a minute             blissful_newton
f859888a7475   moby/buildkit:v0.11.0-rc2                                                "buildkitd --allow-i…"   10 days ago          Up 10 days                    linuxkit-builder
```

then : 
```
docker exec -it lfedge/eve-vtpm:8979f64dd03931c6a6453b7b30ba61e2ad69ef28-dirty-95d5ec3 /bin/sh
Error response from daemon: No such container: lfedge/eve-vtpm:8979f64dd03931c6a6453b7b30ba61e2ad69ef28-dirty-95d5ec3
```

I'm out of ideas.

Reply:
That's your image. Your container ID is 37af2d2fb908

Reply:
Did you try the containerd logging?